API-server panic while trying to run test with TestSource

**Describe the bug**
API-server panic while trying to run test with TestSource

**To Reproduce**
Steps to reproduce the behavior:
1. Create TestSource and Test using it - for example:
```
# postman-executor-smoke-testsource-git-file - TestSource
apiVersion: tests.testkube.io/v1
kind: TestSource
metadata:
  name: testsource-postman-executor-smoke-testsource-git-file
  namespace: testkube
spec:
  type: git-file
  repository:
    type: git
    uri: https://github.com/kubeshop/testkube.git
    branch: main
    path: test/postman/executor-smoke/postman-executor-smoke.postman_collection.json
---
# postman-executor-smoke-testsource-git-file - Test
apiVersion: tests.testkube.io/v3
kind: Test
metadata:
  name: postman-executor-smoke-testsource-git-file
  namespace: testkube
spec:
  type: postman/collection
  source: testsource-postman-executor-smoke-testsource-git-file
```
2. Try to run the test
3. API-server will crash with:
````
{"level":"info","ts":1666098558.8147976,"caller":"event/emitter.go:126","msg":"event published","id":"929b5597-8347-4c92-b1af-eece03ee9b8c","type":"start-testsuite","executionId":"634ea57e87d809fd5e46627e","executionName":"ts-executor-postman-smoke-tests-13","labels":"app=testkube ","error":null}
{"level":"info","ts":1666098558.8168507,"caller":"slacknotifier/slack.go:297","msg":"Execution changed","status":"running"}
{"level":"info","ts":1666098558.8182228,"caller":"event/emitter.go:172","msg":"listener notified","listen-on":["start-test","start-testsuite","end-test-success","end-testsuite-success","end-test-failed","end-testsuite-failed"],"queue-group":"websocket.allevents","selector":"","metadata":{"clients":"[]","events":"[start-test start-testsuite end-test-success end-testsuite-success end-test-failed end-testsuite-failed]","name":"websocket.allevents","selector":""},"id":"929b5597-8347-4c92-b1af-eece03ee9b8c","type":"start-testsuite","executionId":"634ea57e87d809fd5e46627e","executionName":"ts-executor-postman-smoke-tests-13","labels":"app=testkube "}
{"level":"info","ts":1666098558.8177795,"caller":"scheduler/testsuite_scheduler.go:167","msg":"executing testvariablesmap[]request{executor-postman-smoke-tests-postman-executor-smoke-testsource-git-file executor-postman-smoke-tests 0 map[]   map[]   [] []  [] map[] map[] true   0 map[]}","type":"executeTest","testSuiteName":"executor-postman-smoke-tests","name":"run:/postman-executor-smoke-testsource-git-file"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1778a45]

goroutine 932 [running]:
github.com/kubeshop/testkube/pkg/scheduler.mergeContents(0x8?, {{0xc0011bb478, 0x8}, 0xc00124be00, {0x0, 0x0}, {0x0, 0x0}})
        /build/pkg/scheduler/test_scheduler.go:317 +0x45
github.com/kubeshop/testkube/pkg/scheduler.(*Scheduler).getExecuteOptions(_, {_, _}, {_, _}, {{0xc000ee7220, 0x49}, {0xc000bcd580, 0x1c}, 0x3, ...})
        /build/pkg/scheduler/test_scheduler.go:221 +0x17b
github.com/kubeshop/testkube/pkg/scheduler.(*Scheduler).executeTest(_, {_, _}, {{0xc000ecb1a0, 0x2a}, {0x0, 0x0}, {0x0, 0x0}, 0x0, ...}, ...)
        /build/pkg/scheduler/test_scheduler.go:64 +0x486
github.com/kubeshop/testkube/pkg/scheduler.(*Scheduler).executeTestStep(_, {_, _}, {{0xc000b13de8, 0x18}, {0xc000f2b0b0, 0x22}, 0xc00021afe0, 0xc00015e140, 0x0, ...}, ...)
        /build/pkg/scheduler/testsuite_scheduler.go:168 +0x710
github.com/kubeshop/testkube/pkg/scheduler.(*Scheduler).executeTestSuite.func1(0xc00108f080, {{0xc000f2b0b0, 0x22}, 0xd, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0, ...})
        /build/pkg/scheduler/testsuite_scheduler.go:95 +0x4fc
created by github.com/kubeshop/testkube/pkg/scheduler.(*Scheduler).executeTestSuite
        /build/pkg/scheduler/testsuite_scheduler.go:69 +0x805
````

**Expected behavior**
Test should be started correctly, or error message should be returned.

**Version / Cluster**
Client Version 1.6.16
Server Version v1.6.16

--- a/pkg/scheduler/test_scheduler.go
+++ b/pkg/scheduler/test_scheduler.go
@@ -218,7 +218,7 @@ func (s *Scheduler) getExecuteOptions(namespace, id string, request testkube.Exe
 		if err != nil {
 			return options, errors.Errorf("cannot get test source custom resource: %v", err)
 		}
-		testCR.Spec.Content = mergeContents(testCR.Spec.Content, testSourceCR.Spec)
+		testCR.Spec = mergeContents(testCR.Spec, testSourceCR.Spec)
 	}
 
 	test := testsmapper.MapTestCRToAPI(*testCR)
@@ -313,52 +313,56 @@ func mergeEnvs(envs1 map[string]string, envs2 map[string]string) map[string]stri
 	return envs
 }
 
-func mergeContents(testContent *testsv3.TestContent, testSource testsourcev1.TestSourceSpec) *testsv3.TestContent {
-	testContent.Type_ = testSource.Type_
+func mergeContents(test testsv3.TestSpec, testSource testsourcev1.TestSourceSpec) testsv3.TestSpec {
+	if test.Content == nil {
+		test.Content = &testsv3.TestContent{}
+	}
+
+	test.Content.Type_ = testSource.Type_
 	if testSource.Data != "" {
-		testContent.Data = testSource.Data
+		test.Content.Data = testSource.Data
 	}
 
 	if testSource.Uri != "" {
-		testContent.Uri = testSource.Uri
+		test.Content.Uri = testSource.Uri
 	}
 
 	if testSource.Repository != nil {
-		if testContent.Repository == nil {
-			testContent.Repository = &testsv3.Repository{}
+		if test.Content.Repository == nil {
+			test.Content.Repository = &testsv3.Repository{}
 		}
 
-		testContent.Repository.Type_ = testSource.Repository.Type_
-		testContent.Repository.Uri = testSource.Repository.Uri
+		test.Content.Repository.Type_ = testSource.Repository.Type_
+		test.Content.Repository.Uri = testSource.Repository.Uri
 
 		if testSource.Repository.Branch != "" {
-			testContent.Repository.Branch = testSource.Repository.Branch
+			test.Content.Repository.Branch = testSource.Repository.Branch
 		}
 
 		if testSource.Repository.Commit != "" {
-			testContent.Repository.Commit = testSource.Repository.Commit
+			test.Content.Repository.Commit = testSource.Repository.Commit
 		}
 
 		if testSource.Repository.Path != "" {
-			testContent.Repository.Path = testSource.Repository.Path
+			test.Content.Repository.Path = testSource.Repository.Path
 		}
 
 		if testSource.Repository.UsernameSecret != nil {
-			testContent.Repository.UsernameSecret = &testsv3.SecretRef{
+			test.Content.Repository.UsernameSecret = &testsv3.SecretRef{
 				Name: testSource.Repository.UsernameSecret.Name,
 				Key:  testSource.Repository.UsernameSecret.Key,
 			}
 		}
 
 		if testSource.Repository.TokenSecret != nil {
-			testContent.Repository.TokenSecret = &testsv3.SecretRef{
+			test.Content.Repository.TokenSecret = &testsv3.SecretRef{
 				Name: testSource.Repository.TokenSecret.Name,
 				Key:  testSource.Repository.TokenSecret.Key,
 			}
 		}
 	}
 
-	return testContent
+	return test
 }
 
 // TODO: generics

